contents/ruby/Ruby.md

```diff
@@ -0,0 +1,142 @@
+It is widely used across many fields of computer science and software engineering, and its most prevalent use is in web development.
+
+Ruby boasts high programmer productivity with its concise, elegant, and human-readable syntax, which is widely agreed to outweigh the loss in execution speed (as compared to compiled languages such as C and Java) unless the latter is critical.
+
+> Often people, especially computer engineers, focus on the machines. They think, "By doing this, the machine will run faster. By doing this, the machine will run more effectively. By doing this, the machine will something something something."
+> They are focusing on machines. But in fact, we need to focus on humans, on how humans care about doing programming or operating the application of the machines. We are the masters. They are the slaves.
+> 
+> *Yukihiro Matsumoto, creator of Ruby language* 
+
+Popular websites running on Ruby (or its framework): GitHub, Airbnb, Twitter, Shopify.
+
+## Getting Started
+
+[The official website of Ruby](https://www.ruby-lang.org/en/) will have everything you need from release notes to downloads to documentations.
+
+To install Ruby, go to the download page of the above website and follow the instructions.
+If you're a Mac user, you don't need to do anything; Macs come with Ruby pre-installed!
+
+For beginners and intermediate users, [tutorialspoint](https://www.tutorialspoint.com/ruby/index.htm) has great introductory Ruby tutorials, and [rubylearning.com](http://rubylearning.com/satishtalim/tutorial.html) is another great place.
+We will explore some of the more prominent language features of Ruby.
+
+### Everything is an object
+
+Ruby has no concept of "primitives". Anything that can be assigned to a variable is an object. Even numbers and boolean values `true` and `false`.
+
+```rb
+5.times do
+  puts "Hello world!"
+end
+
+# Hello world!
+# Hello world!
+# Hello world!
+# Hello world!
+# Hello world!
+```
+
+as such it becomes very easy to introduce new functions to those "primitives":
+
+```rb
+class Fixnum # integers in Ruby belong to this class
+  def next()
+    return self + 1
+  end
+end
+
+7.next()
+#  => 8
+```
+
+### Readability is king
+
+> Ruby is simple in appearance, but is very complex inside, just like our human body.
+> 
+> *Yukihiro Matsumoto*
+
+Many languages attempt to make their codes as close to pseudocode as possible, but Ruby takes it to another level: Ruby code flows almost as smoothly as an English literary text.
+
+```rb
+for num in 0..5
+  puts "I am counting #{num}!"
+end
+
+unless number.even?
+  puts "Number is odd!"
+end
+
+["chocolate", "strawberry", "vanilla"].each { |flavour|
+  make_ice_cream(flavour)
+}
+```
+
+*No, you're not reading an English poem. You're reading Ruby!*
+
+### Functional programming is encouraged
+
+Ruby supports **map**, **filter**, and **fold-left**.
```

Comment:
Might want to link to Dickson's functional programming book chapter, so that readers that do not know the terms can learn more about it.